File: evidence/generated-systems/golang/aerospace-on-golang/aerospace-on-golang/internal/model/ConnectedAircraft.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// ConnectedAircraft Declaration
//==============================================================
type ConnectedAircraft struct {
    gorm.Model
     CommunicationsProvider                                    string
    AircraftId         *uint
    Aircraft           *Aircraft `gorm:"foreignKey:AircraftId"`
     FlightHealthEvents           []FlightHealthEvent `gorm:"foreignKey:FlightHealthEventsFromConnectedAircraftId"`
     SoftwareLoads           []SoftwareLoad `gorm:"foreignKey:SoftwareLoadsFromConnectedAircraftId"`
    ConnectivityStatus                      ConnectivityStatus

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/aerospace-on-rails/aerospaceonrails/app/models/connectedaircraft.rb
class ConnectedAircraft < ApplicationRecord
  enum ConnectivityStatus: [:Offline, :Online, :Degraded]


  has_many :Aircraft, class_name: 'Aircraft'
  has_many :FlightHealthEvents, class_name: 'FlightHealthEvent'
  has_many :SoftwareLoads, class_name: 'SoftwareLoad'

end
